core/src/main/java/com/orientechnologies/orient/core/schedule/OScheduledEvent.java

```diff
@@ -156,20 +119,25 @@ public Date getStartTime() {
   }
 
   public boolean isRunning() {
-    return this.isRunning;
+    return this.running.get();
   }
 
   public OScheduledEvent schedule() {
-    if (timer != null)
-      timer.cancel();
-
-    bindDb();
-
-    nextExecutionId = getNextExecutionId() + 1;
-    timer = Orient.instance().scheduleTask(new ScheduledTimer(), cron.getNextValidTimeAfter(new Date()), 0);
+    if (isRunning()) {
+      interrupt();
+    }
+    internalSchedule(new ScheduledTimerTask(this));
     return this;
   }
 
+  private void internalSchedule(ScheduledTimerTask timerTask) {
+    synchronized (this) {
+      nextExecutionId.incrementAndGet();
+      timer = Orient.instance().scheduleTask(timerTask, cron.getNextValidTimeAfter(new Date()), 0);
```

Comment:
could you schedule the using the scheduler in the OrientDBInternal instance instead on the Orient singlethon?
as well be aware that the scheduleTask in the singlethon get a runnable and return a TimerTask, so no need to pass in a TimerTask itself, in the new Implementation you need to pass in the TimerTask, as well could you use the `OrientDBInternal.execute(db,user, function)`  for executing the logic (with "admin" user for now) ? this will remove the need of the ThreadLocalFactory.

Reply:
Sure. I will do this today or tommorrow